components/delete_post_modal/delete_post_modal.jsx

```diff
@@ -1,106 +1,83 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See License.txt for license information.
 
+import PropTypes from 'prop-types';
 import React from 'react';
 import {Modal} from 'react-bootstrap';
 import {FormattedMessage} from 'react-intl';
 
-import {deletePost} from 'actions/post_actions.jsx';
-import ModalStore from 'stores/modal_store.jsx';
-import Constants from 'utils/constants.jsx';
 import * as UserAgent from 'utils/user_agent.jsx';
 
-var ActionTypes = Constants.ActionTypes;
+export default class DeletePostModal extends React.PureComponent {
+    static propTypes = {
 
-export default class DeletePostModal extends React.Component {
-    constructor(props) {
-        super(props);
+        post: PropTypes.object.isRequired,
+        commentCount: PropTypes.number.isRequired,
 
-        this.state = {
-            show: false,
-            post: null,
-            commentCount: 0,
-            error: '',
-        };
-    }
+        /**
+         * Does the post come from RHS mode
+         */
+        isRHS: PropTypes.bool.isRequired,
 
-    componentDidMount() {
-        ModalStore.addModalListener(ActionTypes.TOGGLE_DELETE_POST_MODAL, this.handleToggle);
-    }
+        /**
+        * Function called when modal is dismissed
+        */
+        onHide: PropTypes.func.isRequired,
 
-    componentWillUnmount() {
-        ModalStore.removeModalListener(ActionTypes.TOGGLE_DELETE_POST_MODAL, this.handleToggle);
-    }
+        actions: PropTypes.shape({
 
-    componentDidUpdate(prevProps, prevState) {
-        if (this.state.show && !prevState.show) {
-            setTimeout(() => {
-                if (this.deletePostBtn) {
-                    this.deletePostBtn.focus();
-                }
-            }, 200);
-        }
+            /**
+            * Function called for deleting post
+            */
+            deletePost: PropTypes.func.isRequired,
+        }),
     }
 
-    handleDelete = () => {
-        deletePost(
```

Comment:
Thanks @GitHubJasper!
Forgot to mention earlier:
- the `deletePost` at https://github.com/mattermost/mattermost-webapp/blob/master/actions/post_actions.jsx#L205 is of no use now. It should be deleted but make sure to consider moving [these codes](https://github.com/mattermost/mattermost-webapp/blob/master/actions/post_actions.jsx#L234-L242), maybe like so:
```
handleDelete = async () => {
    const {
        actions,
        channelName,
        focusedPostId,
        post,
        teamURL,
    } = this.props;

    const {data} = await actions.deletePost(post);

    if (post.id === focusedPostId && channelName) {
        browserHistory.push(teamURL + '/channels/' + channelName);
    }

    if (data) {
        this.onHide();
    }
}
```
Then at `delete_post_modal/index.js`, need to generate those new props, like so:
```
import {getChannel} from 'mattermost-redux/selectors/entities/channels';
import {getCurrentTeamUrl} from 'mattermost-redux/selectors/entities/teams';

function mapStateToProps(state, ownProps) {
    const channel = getChannel(state, ownProps.post.channel_id);
    let channelName = '';
    if (channel) {
        channelName = channel.name;
    }

    const {focusedPostId} = state.views.channel;

    return {
        channelName,
        focusedPostId,
        teamURL: getCurrentTeamUrl(state),
    };
}
```
(I have not tested above code but more or less it should be like that. Feel free to test functionality :) )